sc-tools/sc-server/src/sc-server-impl/sc-memory-json/sc-memory-json-event/sc_memory_json_events_handler.cpp

```diff
@@ -81,12 +81,8 @@ ScMemoryJsonPayload ScMemoryJsonEventsHandler::HandleGenerate(
     ScAddr const & subscriptionElementAddr = ScAddr(atom["addr"].get<size_t>());
 
     auto const & it = m_deprecatedEventsIdtfsToSystemEventsIdtfs.find(eventClass);
-    if (it == m_deprecatedEventsIdtfsToSystemEventsIdtfs.cend())
-    {
-      errorsPayload = "Unknown sc-event type with system identifier `" + eventClass + "`.";
-      return responsePayload;
-    }
-    eventClass = it->second;
+    if (it != m_deprecatedEventsIdtfsToSystemEventsIdtfs.cend())
+      eventClass = it->second;
 
     ScAddr const & eventClassAddr = m_context->SearchElementBySystemIdentifier(eventClass);
     if (!eventClassAddr.IsValid())
```

Comment:
for line 94:
First of all, it's not safe. We need more checks on input arguments because there are more of them. Secondly, it is correct from the OOP point of view. Constructors should not throw exceptions. Third, it is correct from the point of view of the architecture we use in the sc-machine. The ScAgentContext is a facade over all possible objects used by agents.

Reply:
1. How else should the input arguments be handled? Maybe you talk about that the found node should belong to the event class. Yes, I'll add that
2. Constructor on line 94 doesn't throw any exceptions
3. sc-server uses ScMemoryContext. If you make it use ScAgentContext, you need to refactor the code a little bit